libevmasm/SemanticInformation.cpp

```diff
@@ -278,12 +281,41 @@ bool SemanticInformation::invalidatesStorage(Instruction _instruction)
 	case Instruction::CREATE:
 	case Instruction::CREATE2:
 	case Instruction::SSTORE:
-		return true;
+		return yul::SideEffects::Write;
+
+	case Instruction::SLOAD:
+		return yul::SideEffects::Read;
+
 	default:
-		return false;
+		return yul::SideEffects::Pure;
 	}
 }
 
+yul::SideEffects::Effect SemanticInformation::otherState(Instruction _instruction)
+{
+	switch (_instruction)
+	{
+	case Instruction::CALL:
+	case Instruction::CALLCODE:
+	case Instruction::DELEGATECALL:
+	case Instruction::CREATE:
+	case Instruction::CREATE2:
+	case Instruction::SSTORE:
```

Comment:
Please add a comment:
 - strictly, log0-log4 write to the state, but the EVM cannot read it, so they are just marked as "has other side effects".

Add:
 - selfdestruct